### DISCUSSION TITLE
How do I package up a set of CMake functions to be distributed via VCPKG?

### DISCUSSION THREAD

**Discussion by Cazadorro** [microsoft/vcpkg#35899] [Q&A] [answered]
I have a set of convenience CMake functions and variables that I'd like to use across several different projects.  Right now I'm forced to manually copy each file or use a git submodule to use them.  I could fetch content, but my understanding is that anything like that breaks --only-download and other offline usages of VCPKG, since those cmake commands wouldn't be run until package build time (which is too late to fetch things from the internet).  I would like to instead see if I could use VCPKG to do this,  I see 

https://github.com/microsoft/vcpkg/tree/cb1e4728779e35c4bb60c25114a59c93b3e9be9a/ports/vcpkg-cmake-config

`vcpkg-cmake-config` and friends are basically what this is (though I'm not sure I can access these anywhere but inside of a portfile?).  You add things like `vcpkg-cmake-config` to get access to certain cmake functions inside of the portfie.cmake. I would basically like to be able to do that, but * actually use my own functions in a CMakeLists.txt* that is utilize vcpkg packages, and *hopefully* as if they were normal cmake files to include (ie `include(my-cmake-function-file.cmake)`). 

Is there a method to accomplish this?

**Comment by dg0yt** [comment, marked as answer]
Provide your functions as a CMake config package.
- Let's assume your port is called `my-scripts`.
- Let the port install your script functions as `share/my-scripts/my-scripts-config.cmake`.
- In your CMake projects, simply do `find_package(my-scripts CONFIG REQUIRED)`

And this basically works even without vcpkg - just offer the config file in a location searched by `find_package`.
